gensim/models/ldamodel.py

```diff
@@ -1027,14 +1037,16 @@ def do_mstep(self, rho, other, extra_pass=False):
         logger.debug("updating topics")
         # update self with the new blend; also keep track of how much did
         # the topics change through this update, to assess convergence
-        diff = np.log(self.expElogbeta)
+        previous_Elogbeta = self.state.get_Elogbeta()
         self.state.blend(rho, other)
-        diff -= self.state.get_Elogbeta()
-        self.sync_state()
+
+        current_Elogbeta = self.state.get_Elogbeta()
+        self.sync_state(current_Elogbeta)
 
         # print out some debug info at the end of each EM iteration
         self.print_topics(5)
-        logger.info("topic diff=%f, rho=%f", np.mean(np.abs(diff)), rho)
+        diff = mean_absolute_difference(previous_Elogbeta.ravel(), current_Elogbeta.ravel())
```

Comment:
is `ravel` almost free (i.e. no copy/slowdown here)?

Reply:
I have already used this trick in my one of previous PR.
I made a benchmark then. `.ravel()` is not a free at all but it's the best what I found then.

```ipython
import numpy as np
from gensim.matutils import mean_absolute_difference as mad

a = np.random.random((100, 100))
b = np.random.random((100, 100))

%timeit np.mean(np.abs(a - b))

%timeit mad(a.ravel(), b.ravel())

%timeit mad(a.flatten(), b.flatten())

a_r = a.ravel()
b_r = b.ravel()

%timeit mad(a_r, b_r)
```

For my computer I get;

```
P:\_Projects\gensim\test_ft>ipython  mad.ipy
390 µs ± 12 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
51 µs ± 575 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
145 µs ± 6.5 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
44.3 µs ± 383 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```